Client/Frontend/Browser/Onboarding/OnboardingWebViewController.swift

```diff
@@ -0,0 +1,279 @@
+            return configuration
+        }()
+        
+        return WKWebView(frame: .zero, configuration: configuration)
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let stackView = UIStackView().then {
+            $0.axis = .vertical
+        }
+        
+        [toolbar, webView].forEach(stackView.addArrangedSubview(_:))
+        
+        view.addSubview(stackView)
+        stackView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        webView.navigationDelegate = self
+        webView.load(URLRequest(url: URL(string: "https://brave.com/terms-of-use/")!))
+        
+        toolbar.exitButton.addTarget(self, action: #selector(onExit), for: .touchUpInside)
+        toolbar.backButton.addTarget(self, action: #selector(onBack), for: .touchUpInside)
+        toolbar.forwardButton.addTarget(self, action: #selector(onForward), for: .touchUpInside)
+        toolbar.actionButton.addTarget(self, action: #selector(onAction), for: .touchUpInside)
+    }
+    
+    @objc
+    private func onExit() {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    @objc
+    private func onBack() {
+        if webView.canGoBack {
+            webView.goBack()
+            updateWebPageDomain()
+        }
+        
+        toolbar.backButton.isEnabled = webView.canGoBack
+        toolbar.backButton.tintColor = webView.canGoBack ? UX.buttonEnabledColor : UX.buttonDisabledColor
+    }
+    
+    @objc
+    private func onForward() {
+        if webView.canGoForward {
+            webView.goForward()
+            updateWebPageDomain()
+        }
+        
+        toolbar.forwardButton.isEnabled = webView.canGoForward
+        toolbar.forwardButton.tintColor = webView.canGoForward ? UX.buttonEnabledColor : UX.buttonDisabledColor
+    }
+    
+    @objc
+    private func onAction() {
+        print("Access Button Pressed..")
+    }
+    
+    private func updateWebPageDomain() {
+        toolbar.urlLabel.text = webView.url?.host
+    }
+    
+    private func updateWebPageSecurity() {
+        if let trust = webView.serverTrust {
+            toolbar.secureIcon.isHidden = false
+            
+            var result: SecTrustResultType = .invalid
+            SecTrustEvaluate(trust, &result)
+            
+            if result == .proceed || result == .unspecified {
+                toolbar.secureIcon.tintColor = UX.secureWebPageColor
+                toolbar.urlLabel.textColor = UX.secureWebPageColor
+            } else {
+                toolbar.secureIcon.tintColor = UX.insecureWebPageColor
+                toolbar.urlLabel.textColor = UX.insecureWebPageColor
+            }
+        } else {
+            toolbar.secureIcon.isHidden = true
+            toolbar.urlLabel.textColor = UX.unknownWebPageColor
+        }
+    }
+    
+    private func updateBackForwardUI() {
+        toolbar.backButton.isEnabled = webView.canGoBack
```

Comment:
Remove all the delegate methods below that are implemented simply to update UI and just subscribe to relevant KVO as BrowserViewController does

Reply:
Fixed.